app/api/auth/headersMiddleware.ts

```diff
@@ -0,0 +1,12 @@
+import { Request, Response, NextFunction } from 'express';
+
+export default (req: Request, res: Response, next: NextFunction) => {
+  if (!['POST', 'DELETE', 'PUT', 'OPTIONS'].includes(req.method)) {
+    return next();
+  }
+  if (req.get('X-Requested-With') === 'XMLHttpRequest') {
+    return next();
+  }
+  res.status(401);
```

Comment:
401 means Unauthorized, maybe look if there is a more fitting status code for this case ?

Reply:
Is `400 Bad Request` okay?

Reply:
i do not know, can you check what is that people is using in other apps when doing this kind of checks ? maybe 403 makes more sense ?

Reply:
Will go with that then